tests/pyrog/api/test_column.py

```diff
@@ -0,0 +1,78 @@
+import pytest
+from faker import Faker
+
+from django.urls import reverse
+
+faker = Faker()
+
+pytestmark = pytest.mark.django_db
+
+
+@pytest.mark.parametrize("table, column_field, status_code", [(faker.word(), faker.word(), 201)])
+def test_create_column(
+    api_client,
+    join,
+    input_factory,
+    owner,
+    table,
+    column_field,
+    status_code,
+):
+    url = reverse("columns-list")
+
+    data = {
+        "join": join.id,
+        "input": input_factory().id,
+        "owner": owner.id,
+        "table": table,
+        "column": column_field,
+    }
+    response = api_client.post(url, data)
+
+    assert response.status_code == status_code
+
+
+@pytest.fixture(params=[True, False], ids=["With join", "Without join"])
+def x_column(request, column_factory):
+    return column_factory(with_join=request.param)
+
+
+def test_retrieve_column(api_client, x_column):
+    url = reverse("columns-detail", kwargs={"pk": x_column.id})
+
+    response = api_client.get(url)
+
+    assert response.status_code == 200
+
+
+def test_list_columns(api_client, column_factory):
+    url = reverse("columns-list")
+    column_factory.create_batch(3, with_join=False)
+
+    response = api_client.get(url)
+
+    assert response.status_code == 200
+    assert len(response.data) == 3
+
+
+@pytest.mark.parametrize("table, column_field, status_code", [(faker.word(), faker.word(), 200)])
+def test_update_column(api_client, column, table, column_field, status_code):
+    url = reverse("columns-detail", kwargs={"pk": column.id})
+
+    data = {}
+    if column_field:
```

Comment:
this would make sense if you parametrize the test with an empty `column_field` right ?

Reply:
Sure there is surely more test cases to parametrize, to cover at least all valid inputs

Reply:
ok so it's for later, i'm fine with that :+1: